Zend/zend_inheritance.c

```diff
@@ -570,6 +572,19 @@ static inheritance_status zend_is_class_subtype_of_type(
 	if (have_unresolved) {
 		return INHERITANCE_UNRESOLVED;
 	}
+
+	// replacing static with self in final classes is okay
+	if (!is_intersection && (fe_scope->ce_flags & ZEND_ACC_FINAL) &&
+		(proto_type_full_mask & MAY_BE_STATIC) && instanceof_function(fe_scope, proto_scope)) {
```

Comment:
Likewise, shouldn't this use `unlinked_instanceof` ?
Perhaps it's good to test with an unlinked instance too somehow.